src/gwt/src/org/rstudio/studio/client/workbench/views/source/Source.java

```diff
@@ -586,17 +598,57 @@ public void onNewWorkingCopy(NewWorkingCopyEvent event)
          }
       });
       
-      events.addHandler(PopoutDocEvent.TYPE, this);
-      events.addHandler(DocWindowChangedEvent.TYPE, this);
-      events.addHandler(DocTabDragInitiatedEvent.TYPE, this);
-      events.addHandler(PopoutDocInitiatedEvent.TYPE, this);
-      events.addHandler(DebugModeChangedEvent.TYPE, this);
-      events.addHandler(ReplaceRangesEvent.TYPE, this);
-      events.addHandler(GetEditorContextEvent.TYPE, this);
-      events.addHandler(SetSelectionRangesEvent.TYPE, this);
-      events.addHandler(OpenProfileEvent.TYPE, this);
-      events.addHandler(RequestDocumentSaveEvent.TYPE, this);
-      events.addHandler(RequestDocumentCloseEvent.TYPE, this);
+      events_.addHandler(PopoutDocEvent.TYPE, this);
+      events_.addHandler(DocWindowChangedEvent.TYPE, this);
+      events_.addHandler(DocTabDragInitiatedEvent.TYPE, this);
+      events_.addHandler(PopoutDocInitiatedEvent.TYPE, this);
+      events_.addHandler(DebugModeChangedEvent.TYPE, this);
+      events_.addHandler(ReplaceRangesEvent.TYPE, this);
+      events_.addHandler(GetEditorContextEvent.TYPE, this);
+      events_.addHandler(SetSelectionRangesEvent.TYPE, this);
+      events_.addHandler(OpenProfileEvent.TYPE, this);
+      events_.addHandler(RequestDocumentSaveEvent.TYPE, this);
+      events_.addHandler(RequestDocumentCloseEvent.TYPE, this);
+      
+      initialized_ = true;
+   }
+
+   public void loadFullSource()
+   {
+      AceEditor.preload();
```

Comment:
This might not be safe -- if I understand correctly, this call will load Ace concurrently, but it seems like since we're about to initialize tabs (which might require Ace to be available) we would want to ensure Ace has really been loaded.

Reply:
Good catch. I updated the code with a new method that loads Ace and calls loadFullSource (now private) in its callback.